Microsoft.Research/Contracts/WindowsBase/System.Windows.Threading.DispatcherObject.cs

```diff
@@ -22,7 +22,26 @@ public abstract class DispatcherObject
   {
     protected DispatcherObject() { }
 
-    public Dispatcher Dispatcher { get { Contract.Ensures(Contract.Result<Dispatcher>() != null); return null; } }
+    // DetachFromDispatcher is e.g. called by Freezable.Freeze()
+    //
+    // This method allows certain derived classes to break the dispatcher affinity
+    // of our objects.
+    // [FriendAccessAllowed] // Built into Base, also used by Framework.
+    internal void DetachFromDispatcher()
```

Comment:
:bulb: Thanks for clarifying. When I was reading through the code in the current form, I didn't feel like I understood exactly why `Dispatcher` could return null. If the comment for `Dispatcher` was updated in the way I mentioned in my other comment (on that property), I feel like this internal method and the preceding comment(s) could be removed from the code without reducing the clarity of this set of contracts, as freezing a `Freezable` is a well-understood concept for API consumers.